_rfcs/content-signature.md

```diff
@@ -0,0 +1,121 @@
+>
+>The Content-MD5 header field MAY be generated by an origin server or client to function as an integrity check of the entity-body. Only origin servers or clients MAY generate the Content-MD5 header field; proxies and gateways MUST NOT generate it, as this would defeat its value as an end-to-end integrity check. Any recipient of the entity- body, including gateways and proxies, MAY check that the digest value in this header field matches that of the entity-body as received.
+>
+>The MD5 digest is computed based on the content of the entity-body, including any content-coding that has been applied, but not including any transfer-encoding applied to the message-body. If the message is received with a transfer-encoding, that encoding MUST be removed prior to checking the Content-MD5 value against the received entity.
+>
+>This has the result that the digest is computed on the octets of the entity-body exactly as, and in the order that, they would be sent if no transfer-encoding were being applied.
+>
+>HTTP extends RFC 1864 to permit the digest to be computed for MIME composite media-types (e.g., multipart/* and message/rfc822), but this does not change how the digest is computed as defined in the preceding paragraph.
+>
+>There are several consequences of this. The entity-body for composite types MAY contain many body-parts, each with its own MIME and HTTP headers (including Content-MD5, Content-Transfer-Encoding, and Content-Encoding headers). If a body-part has a Content-Transfer- Encoding or Content-Encoding header, it is assumed that the content of the body-part has had the encoding applied, and the body-part is included in the Content-MD5 digest as is -- i.e., after the application. The Transfer-Encoding header field is not allowed within body-parts .
+
+## Reference to http-signatures
+
+The draft standard describes the scheme for a `Signature` header as follows:
+
+>Creating a Signature
+>
+>
+>   In order to create a signature, a client MUST:
+>
+>   1.  Use the contents of the HTTP message, the `headers` value, and
+>       the Signature String Construction algorithm to create the
+>       signature string.
+>
+>   2.  The `algorithm` and key associated with `keyId` must then be used
+>       to generate a digital signature on the signature string.
+>
+>   3.  The `signature` is then generated by base 64 encoding the output
+>       of the digital signature algorithm.
+
+## Synthesising `Content-Signature`
+
+If we take the insights from both of these standards we can synthesise a new header, `Content-Signature`. 
+
+The digest part of the signature is computed as per `Content-MD5` over the entity body bytes using the algorithm specified, before any transfer encoding (such as compression) is applied.
+
+This digest is then signed and the data about the signature made available in the `Content-Signature` header which correspondingly requires the fields as defined:
+
+* `algorithm`
+* `keyId`
+* `signature`
+
```

Comment:
I would put at the end the final "spec" :)
Content-Signature: keyId="<keyId>",algorithm="<signatureAlgorithm>",signature="<base64EncodedSignature>"